neurom/core/soma.py

```diff
@@ -443,6 +363,127 @@ def _soma_three_point_cylinders_volume(morphio_soma):
     return soma_algo(morphio_soma)
 
 
+def soma_overlaps(soma, points, exclude_boundary=False):
+    """Check if soma overlaps with points."""
+
+    if hasattr(soma, "to_morphio"):
+        morphio_soma = soma.to_morphio()
+    else:
+        morphio_soma = soma
```

Comment:
Maybe we could also accept morphologies as input?
```suggestion
    if hasattr(soma, "soma"):
        input_soma = soma.soma()
    else:
        input_soma = soma
    if hasattr(input_soma, "to_morphio"):
        morphio_soma = input_soma.to_morphio()
    else:
        morphio_soma = input_soma
```

Reply:
Why give that additional responsibility to this function? This is what the respective feature is for.

Reply:
I think it's interesting to make our codes simpler. Especially with #1119 it possible to write something like `soma_center(morph)` which makes sense IMO.

Reply:
I still don't think the soma module should be responsible for handling the morphology. Especially, when passing the soma is as simple as `soma_radius(morph.soma)`

Reply:
I guess that what bothers me is that the names are over explicit. If the functions of the `soma` module can only work on soma objects, why naming the functions `soma.soma_*`? For me, if you specify the object in the name it's because it is not already explicit, which is the case if the function can accept a morphology. But if it only accepts a soma it's already explicit. For example, writing `soma.get_radius(morph.soma)` would make more sense to me than `soma.soma_radius(morph.soma)`. WDYT?

Reply:
No issue with renaming. `soma_*` changed to `get_*`

Reply:
Ok!